lib/internal/per_context/primordials.js

```diff
@@ -65,16 +65,22 @@ function copyPropsRenamedBound(src, dest, prefix) {
 
 function copyPrototype(src, dest, prefix) {
   for (const key of Reflect.ownKeys(src)) {
-    if (typeof key === 'string') {
-      const desc = Reflect.getOwnPropertyDescriptor(src, key);
-      if (typeof desc.value === 'function') {
-        desc.value = uncurryThis(desc.value);
-      }
-      Reflect.defineProperty(
-        dest,
-        `${prefix}${key[0].toUpperCase()}${key.slice(1)}`,
-        desc);
+    let newKey;
+    if (typeof key === 'symbol') {
+      const name = key.description.slice(7);
+      newKey = `${prefix}Symbol${name[0].toUpperCase()}${name.slice(1)}`;
+    } else {
+      newKey = `${prefix}${key[0].toUpperCase()}${key.slice(1)}`;
     }
+    const desc = Reflect.getOwnPropertyDescriptor(src, key);
+    if (typeof desc.value === 'function') {
+      desc.value = uncurryThis(desc.value);
+    } else if (typeof desc.get === 'function') {
+      desc.value = uncurryThis(desc.get);
+      delete desc.get;
+      delete desc.set;
+    }
+    Reflect.defineProperty(dest, newKey, desc);
```

Comment:
We should probably also save the setter if it exists.

```suggestion
    const desc = Reflect.getOwnPropertyDescriptor(src, key);
    let setterDesc;
    if (typeof desc.value === 'function') {
      desc.value = uncurryThis(desc.value);
    } else if (typeof desc.get === 'function') {
      desc.value = uncurryThis(desc.get);
      if (typeof desc.set === 'function') {
        setterDesc = { value: uncurryThis(desc.set), enumerable: desc.enumerable };
      }
      delete desc.get;
      delete desc.set;
    }
    Reflect.defineProperty(dest, newKey, desc);
    if (setterDesc) {
      Reflect.defineProperty(dest, `${newKey}Setter`, desc);
    }
```

Reply:
I’ve extracted this change into <https://github.com/nodejs/node/pull/36329>.